beater/server_test.go

```diff
@@ -162,86 +144,84 @@ func TestServerCORS(t *testing.T) {
 	}
 
 	for idx, test := range tests {
-		apm.server.Handler = newMuxer(
-			&Config{
-				MaxUnzippedSize:     1024 * 1024,
-				ConcurrentRequests:  40,
-				MaxRequestQueueTime: time.Second * 10,
-				Frontend: &FrontendConfig{
-					Enabled:      &true,
-					RateLimit:    10,
-					AllowOrigins: test.allowedOrigins},
-			},
-			nopReporter)
-		rec := httptest.NewRecorder()
-		req, _ := http.NewRequest("POST", FrontendTransactionsURL, bytes.NewReader(testData))
+		ucfg, err := common.NewConfigFrom(m{"frontend": m{"enabled": true, "allow_origins": test.allowedOrigins}})
+		assert.NoError(t, err)
+		apm, teardown := setupServer(t, ucfg)
+		baseUrl, client := apm.client(false)
+		req, err := http.NewRequest("POST", baseUrl+FrontendTransactionsURL, bytes.NewReader(testData))
 		req.Header.Set("Origin", test.origin)
 		req.Header.Set("Content-Type", "application/json")
-		apm.server.Handler.ServeHTTP(rec, req)
-
-		assert.Equal(t, test.expectedStatus, rec.Code, fmt.Sprintf("Failed at idx %v; %s", idx, rec.Body.String()))
+		assert.NoError(t, err)
+		res, err := client.Do(req)
+		assert.Equal(t, test.expectedStatus, res.StatusCode, fmt.Sprintf("Failed at idx %v; %s", idx, body(t, res)))
+		teardown()
 	}
 }
 
 func TestServerNoContentType(t *testing.T) {
 	apm, teardown := setupServer(t, nil)
 	defer teardown()
 
-	rr := httptest.NewRecorder()
-	apm.server.Handler.ServeHTTP(rr, makeTestRequest(t))
-	assert.Equal(t, http.StatusBadRequest, rr.Code, rr.Body.String())
-}
-
-func TestServerSecureUnknownCA(t *testing.T) {
-	apm, teardown := setupServer(t, withSSL(t, "127.0.0.1"))
-	defer teardown()
-
-	_, err := postTestRequest(t, apm.server, nil, "https")
-	assert.Contains(t, err.Error(), "x509: certificate signed by unknown authority")
-}
-
-func TestServerSecureSkipVerify(t *testing.T) {
-	apm, teardown := setupServer(t, withSSL(t, "127.0.0.1"))
-	defer teardown()
-
-	res, err := postTestRequest(t, apm.server, insecureClient(), "https")
-	assert.Nil(t, err)
-	assert.Equal(t, res.StatusCode, http.StatusAccepted)
+	baseUrl, client := apm.client(false)
+	req := makeTransactionRequest(t, baseUrl)
+	res, error := client.Do(req)
+	assert.NoError(t, error)
+	assert.Equal(t, http.StatusBadRequest, res.StatusCode, body(t, res))
 }
 
-func TestServerSecureBadDomain(t *testing.T) {
-	apm, teardown := setupServer(t, withSSL(t, "ELASTIC"))
-	defer teardown()
-
-	_, err := postTestRequest(t, apm.server, nil, "https")
-
-	msgs := []string{
-		"x509: certificate signed by unknown authority",
-		"x509: cannot validate certificate for 127.0.0.1",
+func TestServerSSL(t *testing.T) {
+	tests := []struct {
+		domain           string
+		expectedMsgs     []string
+		insecure         bool
+		statusCode       int
+		overrideProtocol bool
+	}{
+		{
+			domain: "127.0.0.1", expectedMsgs: []string{"x509: certificate signed by unknown authority"},
```

Comment:
what do you think about giving each of these a name to help explain what's being tested?  Extracting from the original `TestServer<name>` would work well.

Reply:
yeah, im starting to question table tests tbh... they are harder to write, to read and to debug, not sure its worth.
done all here 2348d14f12016f001ca5e7cbffd5201c8f67b757